.circleci/config.yml

```diff
@@ -49,14 +51,14 @@ jobs:
               rm -f dot.bin
             fi
 
-      - run:
-          # We use the python Pipfile.lock as the cache key and also
-          # the hash of the current python.
-          # If dependencies haven't changed then use the cache.
-          name: Create Python cache key
+      - run: &create_python_cache_key
+          # Combine hashes of the Python interpreter and Pipfile plus today's
+          # date into a file whose checksum will key the Python environ. cache.
+          name: Create Python environment cache key
           command: |
-            md5sum lib/Pipfile.locks/${CIRCLE_JOB} > ~/python_version.md5
-            md5sum $(which python) >> ~/python_version.md5
+            md5sum $(which python) > ~/python_version.md5
+            md5sum lib/Pipfile >> ~/python_version.md5
+            date +%F >> ~/python_version.md5
```

Comment:
There was a discussion in the latter half of an eng standup this week. Both the old and new ways are valid, but have tradeoffs:

Old way:
+ Dependency versions are committed to the source tree, and thus fixed for CI (but not releases or user/developer installs)
- Every time `lib/Pipfile`is changed, an engineer needs to jump into virtual envs for each Python version we want to test against to regenerate the `Pipfile.lock` for that version. This cost increases linearly with the number of Python versions we test against.
- As new versions of packages we depend against are released, we don't test against those versions until the next time someone regenerates the lock files, even though our users are exposed to those newer versons.
- When someone makes any change to `lib/Pipfile`, *all* dependencies are then potentially updated, batching maybe weeks of failures together (again, users were already exposed to these, since the lockfiles are just for CI, not releases).

New way:
+ No more lockfile generation in every Python version for engineers when making changes to `lib/Pipfile`
+ We regularly follow newer versions of packages we depend on that are compatible with the specifications in `lib/Pipfile`, so we're testing closer to what users would experience, and aren't surprised by breaks that could have already been known the next time we change this.
+ By still caching daily, we don't slow down all CI runs having `pipenv`'s solver run, just one per day.
-  When packages we depend on change in a way that breaks a test, we find out seemingly randomly when the daily cache expires.

Those are the tradeoffs I can think of. @tvst verbally okayed going with the daily caching approach.